### PR TITLE
refactor(experimental): create @solana/rpc-transport-http package

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR extract the HTTP implementation of the `RpcTransport` into its own package.

Notice that it does not include any Solana-specific types or helpers such as `ClusterUrl` since this HTTP transport should be used as a standalone HTTP transport of the `@solana/rpc-spec` package. It is the responsibility of the Solana-specific `@solana/rpc` package — created later on in the stack — to wrap this transport and introduce these concepts.

RPC packages reshuffle@2x (4)

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2147?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2147** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2146** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2145** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2144** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2147?utm_source=stack-comment).

**Review by steveluscher — APPROVED** [review]
This is fine, but one problem about replicating the code here and deleting the other stuff later is that we'll lose the entire git history. If you can nuke it and add it in the same PR so that git recognizes the move, that would be better.

[queuepatrick](https://app.graphite.dev/macros/solana-labs)

**Comment by lorisleiva** [comment]
@steveluscher The only issue is `rpc-transport` contains more than just this code so I'd need to be merging like 4 PRs together. It also means that this combined PR would also need to include the re-wiring of all consumers which would be massive (library, accounts, graphql, etc). I tried to split the stack into more digestible narratives but yeah it's a shame about the git history. 😞

**Comment by steveluscher** [comment]
We can always squash the stack into a single commit when we jam this in.

**Comment by lorisleiva** [comment]
Perfect, let's do that! 👌

File: packages/rpc-transport-http/src/http-transport-errors.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    readonly statusCode: number;
+    constructor(details: SolanaHttpErrorDetails) {
+        super(`HTTP error (${details.statusCode}): ${details.message}`);
+        Error.captureStackTrace(this, this.constructor);
+        this.statusCode = details.statusCode;
+    }
+    get name() {
+        return 'SolanaHttpError';
+    }
+}
```
Comment (buffalojoec) [inline packages/rpc-transport-http/src/http-transport-errors.ts:16]: Should these be here? Or should they be called `Solana`?

From PR description:

> Notice that it does not include any Solana-specific types

**Review by buffalojoec — COMMENTED** [review]
Ohhh I'm starting to like where this is going.

File: packages/rpc-transport-http/src/http-transport-errors.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/rpc-transport-http/src/http-transport-errors.ts:16]: I was going to replace that with the `SolanaError` from `@solana/errors` but we decided to do that outside of this PR so I just left it as-is for now.

**Comment by lorisleiva** [comment]
✅ This PR was merged as part of https://github.com/solana-labs/solana-web3.js/pull/2168.

**Pull request closed without merging**